google-cloud-pubsub/lib/google/cloud/pubsub/subscriber.rb

```diff
@@ -66,20 +69,28 @@ def stop
           synchronize do
             return if @request_queue.nil?
             @request_queue.push self
+            @inventory.clear
```

Comment:
You [commented below](https://github.com/GoogleCloudPlatform/google-cloud-ruby/pull/1542/commits/b6e1096b5ac9b009900d2632dbba8712d4ac0ef0#r123351117) to @remi: 

> Subscriber#stop only halts the pulling of new messages. It does not ensure that all messages have been processed. It updates the state of the Subscriber object and will return quickly. Subscriber#wait! will block until all the current messages in the queue are processed. Because it blocks it has the bang to indicate that it could take a while. 

So I had the idea that user calls `#stop` and `#wait!` to stop retrieving, but then process the inventory.